src/java/org/apache/cassandra/index/sai/disk/v1/IndexWriterConfig.java

```diff
@@ -49,6 +49,9 @@ public class IndexWriterConfig
     public static final String ALPHA = "alpha";
     public static final String ENABLE_HIERARCHY = "enable_hierarchy";
     public static final String SIMILARITY_FUNCTION = "similarity_function";
+    // num of vector in vector memtable index to trigger memtable flush
+    public static final String VECTOR_FLUSH_THRESHOLD = "vector_flush_threshold";
+    public static final String VECTOR_FLUSH_PERIOD_IN_MS = "vector_flush_period_in_ms";
```

Comment:
Because we have generic ORDER BY, I think we should just make these generic settings and then we can implement for the trie index, too. We could use: `flush_threshold_max_rows` and `flush_threshold_max_age_minutes`.

Reply:
We already have a generic table setting for this, why do we need it specific to vectors? https://github.com/datastax/cassandra/blob/main/src/java/org/apache/cassandra/schema/TableParams.java#L96

Reply:
> We already have a generic table setting for this, why do we need it specific to vectors?

It's to flush only tables with vector index.

Reply:
> I think we should just make these generic settings and then we can implement for the trie index, too. We could use: flush_threshold_max_rows and flush_threshold_max_age_minutes.

if one system property works for all index types, I can change it

Reply:
> flush_threshold_max_age_minutes

if memtable index is empty, it will reschedule.. it's not "max age", more like flush period